permissions_engine/permissions.rego

```diff
@@ -56,11 +56,24 @@ controlled_allowed = controlled_access_list[username]{
     valid_token == true                  # extant, valid token
 }
 
+#
+# is it for counts endpoint?
+#
+
+default counts_allowed = false
+
+counts_allowed = true {
+    valid_token == true
+    input.method = "GET"
+    input.path = ["counts"]
+}
+
 #
 # List of all allowed datasets for this request
 #
 
-datasets = array.concat(array.concat(open_datasets, registered_allowed), controlled_allowed) {
-  input.method = "GET"                   # only allow GET requests
-  input.path = ["beacon"]                # only allow queries to beacon endpoint
-}
\ No newline at end of file
+datasets = array.concat(open_datasets, opt_in_datasets) { counts_allowed == true }
```

Comment:
You can write this more idiomatically in rego as:

```
datasets = array.concat(array.concat(open_datasets, registered_allowed), controlled_allowed) {
    input.method = "GET"                   # only allow GET requestst
    input.path = ["beacon"]
}

datasets = array.concat(open_datasets, opt_in_datasets) {
          counts_allowed == true
}
```

or even:

```
datasets = array.concat(array.concat(open_datasets, registered_allowed), controlled_allowed) {
    input.method = "GET"                   # only allow GET requestst
    input.path = ["beacon"]
}

datasets = array.concat(open_datasets, opt_in_datasets) {
     valid_token == true
     input.method = "GET"
     input.path = ["counts"]
}
```

Could you make one of those changes and verify that it works?